safe_core/src/client/mock/vault.rs

```diff
@@ -532,13 +562,17 @@ impl Vault {
             }
             Request::InsAuthKey {
                 key,
-                permissions,
+                token,
                 version,
             } => {
                 let result = if owner_pk != requester_pk {
                     Err(SndError::AccessDenied("Not the owner".to_string()))
                 } else {
-                    self.ins_auth_key(&requester.name(), key, permissions, version)
+                    let serialized_caveats = serialize(&token).map_err(|_| {
+                        SndError::FailedToParse("Error serializing caveats".to_string())
+                    })?;
+                    let hash = tiny_keccak::sha3_256(serialized_caveats.as_slice());
```

Comment:
and only caveats here?

Reply:
No, we are hashing the whole token. Just a wrong variable name :sweat_smile: